stl/inc/mutex

```diff
@@ -60,7 +60,7 @@ public:
         }
     }
 
-    void unlock() {
+    void unlock() noexcept /* strengthened */ {

```

Comment:
I'm noticing that `try_lock` is not similarly marked, and in fact does throw sometimes, despite having the same requirement of "_Throws_: Nothing" in the standard. I am concerned that this might be a bug in our implementation.

Reply:
I suspect that `try_lock` only throws on precondition violation, which would allow us to strengthen, but we need to validate that expectation.

Reply:
I've filed #3324 to investigate the `try_lock` issue so it need not hold up this PR.